backend/app/views.py

```diff
@@ -274,4 +278,34 @@ def corpus(request, corpus_id):
     }
 
     return render(request, 'index.html', context)
-    
+
+
+@api_view(['GET'])
+def all_frequency_analyses(request):
+    freq_analysis_objs = FrequencyAnalysis.objects.all()
+    serializer = FrequencyAnalysisSerializer(freq_analysis_objs, many=True)
+    return Response(serializer.data)
+
+
+@api_view(['POST'])
+def add_frequency_analysis(request):
+    attributes = request.data
+    corpus_id = attributes['corpus_id']
+    gender_ids = attributes['gender_ids']
+    gender_set = Gender.objects.filter(id__in=gender_ids)
+    frequency_entry = FrequencyAnalysis.objects.filter(corpus__id=corpus_id)\
+        .annotate(num_genders=Count('genders')).filter(num_genders=len(gender_ids))
+    for id in gender_ids:
+        frequency_entry = frequency_entry.filter(genders__id=id)
+    if frequency_entry.exists():
+        freq_analysis_obj = frequency_entry.get()
```

Comment:
@phuang00 Going off this "the simplest way is first filtering the instances for ones with the same number of genders and then going through the list of genders and ensuring that each of them is in the analysis instance" . Maybe there can be a boolean that keeps track of if all genders are present, I think that renaming the `frequency_entry` inside the for loop to something else and having a boolean variable may do what you are intending. 
```suggestion
    all_genders_appear = True
    for id in gender_ids:
        gender_frequency = frequency_entry.filter(genders__id=id)
        if not gender_frequency:
            all_genders_appear = False
            break
    if all_genders_appear:
        freq_analysis_obj = frequency_entry.get()
```

Reply:
The check `all_genders_appear` is a little redundant because this part of the code, `.annotate(num_genders=Count('genders')).filter(num_genders=len(gender_ids))` on line 297 ensures that the only entries in the Query Set will only have the same number of genders as `gender_ids` and there will not be repeat genders. Thus, the only way that `frequency_entry.exists()` will return `True` is if all of the genders in `gender_ids` are present in an entry in the `FrequencyAnalysis` model.

Reply:
Also, if we rename the inner loop to `gender_frequency`, we constantly iterate over the same `frequency_entry` so if we have entries in the model whose `genders` fields do not intersect, but the union of them contains everything in the `gender_ids` list, we will get a false positive.  One such example is if we have the genders `[1, 3], [2, 4]` in the database and we are looking for `[1, 2]`. `frequency_entry` will contain both `[1, 3]` and `[2, 4]` so when we call `gender_frequency = frequency_entry.filter(genders__id=id)`, `gender_frequency` will not be empty in either iterations of the for loop and `freq_analysis_obj = frequency_entry.get()` will return an error because `frequency_entry` contains more than one entry in the database.